SConstruct

```diff
@@ -1074,207 +1108,212 @@ env['python_cmd_esc'] = quoted(env['python_cmd'])
-                        "except AttributeError:",
-                        "    print(site.USER_SITE)"))
-
-    if env['python_array_home']:
-        script = "sys.path.append({})\n".format(env['python_array_home']) + script
+        assert cython_version >= cython_min_version
+    except (ImportError, AssertionError):
+        message = ("Cython not found or incompatible version: "
+                   "Found {0} but {1} or newer is required.".format(cython_version,
+                                                                    cython_min_version))
+        if require_python:
+            print('ERROR: ' + message)
+            sys.exit(1)
+        elif want_python:
+            print('WARNING: ' + message)
+            env['python_package'] = 'minimal'
+    else:
+        print('INFO: Using Cython version {0}.'.format(cython_version))
+
+if env['python_package'] in ('full', 'minimal', 'default'):
+    # Check the version of the Python package we want to build
+    script = '\n'.join(["from __future__ import print_function",
+                        "import sys",
+                        "print('{v.major}.{v.minor}'.format(v=sys.version_info))"])
 
     try:
-        info = getCommandOutput(env['python_cmd'], '-c', script)
-    except OSError as err:
-        if env['VERBOSE']:
-            print('Error checking for Python 2:')
+        env['python_version'] = getCommandOutput(env['python_cmd'], '-c', script)
+    except (OSError, subprocess.CalledProcessError) as err:
+        if env['python_package'] in ['full', 'minimal']:
+            print('ERROR: Problem checking for Python:')
             print(err)
-        warn_no_python = True
-    except subprocess.CalledProcessError as err:
-        if env['VERBOSE']:
-            print('Error checking for Python 2:')
-            print(err, err.output)
-        warn_no_python = True
+            sys.exit(1)
+        else:
+            print('WARNING: Problem checking for Python:')
+            print(err)
+            print('Continuing with default parameters.')
     else:
-        (env['python_version'], numpy_version,
-         env['python_usersitepackages']) = info.splitlines()[-3:]
-        numpy_version = LooseVersion(numpy_version)
-        if numpy_version == LooseVersion('0.0.0'):
-            python_message += "NumPy for Python {0} not found.\n".format(env['python_version'])
+        major = env['python_version'][0]
+        py_pkg = 'python{}_package'.format(major)
+        if env[py_pkg] in ['full', 'minimal']:
+            print("ERROR: The python_package points to Python {v} and python{v}_package has been "
+                  "specified. Please change python_package to point to a different version of "
+                  "Python or disable it.".format(v=major))
+            sys.exit(1)
+        elif env[py_pkg] == 'default':
+            py_vars = {'python{}_array_home': '', 'python{}_cmd': 'python{}'.format(major),
+                       'python{}_prefix': '$prefix'}
```

Comment:
Shouldn't these values correspond to the values of `env['python_cmd']` etc.?

Reply:
These are the default values for these variables. I'm checking whether the value present in the `env` is equal to the default. This could be clearer, I'll clean it up.